Add Troubleshooting note in MailChimp (#6417)
segmentio/segment-docs: src/connections/destinations/catalog/mailchimp/index.md

### Why are my calls with trait arrays not showing up in Mailchimp?
Mailchimp doesn't support arrays as traits values. This can cause calls to not show up.

> ### Frequent 404 Bad Requests from Identify events without error message?

Reviewer: ```suggestion
### Why are there frequent 404 Bad Requests from Identify events without an error message?
```

> If you are sending concurrent requests for the same userId, MailChimp will block the events due to how MailChimp restricts each API key to a maximum of 10 concurrent requests. 

Reviewer: ```suggestion
If you send concurrent requests for the same userId, MailChimp blocks the events because MailChimp restricts each API key to a maximum of 10 concurrent requests. 
```
